deploy-packit-pr/README.md

```diff
@@ -0,0 +1,120 @@
+# Deploy a testing instance for Packit-as-a-Service
+
+AC:
+
+* figure out how we want to do this epic
+* write down your findings in a document
+* discuss with the team and get a consensus
+
+## How should the deployment be triggered?
+
+At least in the beginning we might not want to deploy every single PR, so we
+might need to have an explicit way for triggering this.
+
+**Minimal solution:** write a script taking some arguments, which each of us
+can run locally. For this each of us would need to have the credentials to the
+environment(s) the script needs to interact with.
+
+**One step further:** have some service/bot, listening to GitHub
+comments/label changes and running the deployment script. We could check with
+softwarefactory folks if we could get a new Zuul pipeline to handle this.
+
+## What if multiple repositories are involved?
+
+The script/bot should understand references to PRs of the repositories. (Later
+on we can extend this to be able to reference branches or commits or even
+different repositories, but for the beginning we can stick to PRs.)
+
+By default all components should use stg images, but it should be possible to
+modify this default to 'prod'.
+
+Repositories:
+
+* packit/packit-service
+* packit/packit
+* packit/ogr
+* packit/dashboard
+* packit/tokman
+* packit/sandcastle
+* packit/deployment
+* packit/packit-service-centosmsg
+* packit/packit-service-fedmsg
+
+Examples:
+
+```
+$ deploy --base=prod packit-service=614
+$ deploy packit-service=800 ogr=123
+$ deploy packit=456 deployment=321
+```
+
+## What are the steps required to deploy the new content?
+
+1. Build images from the PRs.
+2. Check if the test environment is free.
+3. Make sure the test environment is cleaned up after previous deployments.
+4. Run deployment with the new images.
+
+## Where to build the images?
+
+* Docker Hub
+    - Slow; there can be only one build in our project, since we are using a
+      free account.
+    - It's not that obvious how to trigger on-demand builds from PRs.
+        - Actually, @sakalosj told that it's possible to trigger builds using the
+          Docker Hub API, but the method is not documented.
+* Quay
+    - Similar to Docker Hub.
```

Comment:
since Quay is a service maintained by Red Hat, I'd love to know if we can switch to using it (and do the switch, e.g. in Q4)

Reply:
Filed packit/packit-service#833 to look into this.